### PR TITLE
[New resource] `azurerm_machine_learning_workspace_network_outbound_rule_fqdn`

### PR DESCRIPTION
**Pull request by xuzhang3**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

Add support for Machine Learning Workspace Network Outbound Rule FQDN:
- [x] FQDN (`azurerm_machine_learning_workspace_network_outbound_rule_fqdn`)
- [ ] Service Tag 
- [ ] Private Endpoint

```log
=== RUN   TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== PAUSE TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== RUN   TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
=== PAUSE TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
=== CONT  TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== CONT  TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
--- PASS: TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic (551.62s)
--- PASS: TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update (660.47s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       662.982s
```

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [ ] Bug Fix
- [x] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
looks like some test failures
'
image

**Comment by xuzhang3** [comment]
@katbyte  test  fixed
```log
=== RUN   TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== PAUSE TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== RUN   TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
=== PAUSE TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
=== RUN   TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_requiresImport
=== PAUSE TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_requiresImport
=== CONT  TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_requiresImport
=== CONT  TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic
=== CONT  TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update
--- PASS: TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_requiresImport (541.75s)
--- PASS: TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_basic (554.67s)
--- PASS: TestAccMachineLearningWorkspaceNetworkOutboundRuleFqdn_update (926.93s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       942.843s

```

**Review by katbyte — APPROVED** [review]
LGTM ❄️

File: website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_machine_learning_workspace_network_outbound_rule_fqdn" "example" {
+  name         = "example-outboundrule"
+  workspace_id = azurerm_machine_learning_workspace.example.id
+  destination  = "destination"
```
Comment (katbyte) [inline website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown:70]: this is supposed to be a FQDN?
```suggestion
  destination  = "example.com"
```

File: website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Machine Learning Workspace FQDN Network Outbound Rule. Changing this forces a new resource to be created.
+
+* `workspace_id` - (Required) Specifies the ID of the Machine Learning Workspace. Changing this forces a new resource to be created.
+
+* `destination` - (Required) Specifies the fully qualified domain name to allow for outbound traffic.
```
Comment (katbyte) [inline website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown:82]: should this be
```suggestion
* `destination_fqdn` - (Required) Specifies the fully qualified domain name to allow for outbound traffic.
```
as it is a FQDN?

File: internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_fqdn_resource.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managednetwork.ValidateWorkspaceID,
+		},
+
+		"destination": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```
Comment (katbyte) [inline internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_fqdn_resource.go:64]: should we be validating this is a FQDN?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_fqdn_resource.go (hunk 3 above)
Comment (xuzhang3) [reply internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_fqdn_resource.go:64]: There is no restriction on FQDN in the portal, as long as it is a string, checking that the string should not be empty is enough

File: website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown (hunk 2 above)
Comment (xuzhang3) [reply website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown]: updated

File: website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown (hunk 1 above)
Comment (xuzhang3) [reply website/docs/r/machine_learning_workspace_network_outbound_rule_fqdn.html.markdown]: updated

**Comment by xuzhang3** [comment]
@katbyte  All updated
image

**Review by katbyte — APPROVED** [review]
Thanks LGTM 🦅

**Pull request merged**